dbms/src/IO/Compression/CompressionCodecRunLength.cpp

```diff
@@ -0,0 +1,154 @@
+// Copyright 2023 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <Common/Exception.h>
+#include <IO/Compression/CompressionCodecRunLength.h>
+#include <IO/Compression/CompressionInfo.h>
+#include <IO/Compression/CompressionSettings.h>
+#include <IO/Compression/EncodingUtil.h>
+#include <IO/Compression/ICompressionCodec.h>
+#include <common/unaligned.h>
+#include <lz4.h>
+
+#include <magic_enum.hpp>
+
+
+namespace DB
+{
+
+namespace ErrorCodes
+{
+extern const int CANNOT_COMPRESS;
+extern const int CANNOT_DECOMPRESS;
+} // namespace ErrorCodes
+
+CompressionCodecRunLength::CompressionCodecRunLength(CompressionDataType data_type_)
+    : data_type(data_type_)
+{}
+
+UInt8 CompressionCodecRunLength::getMethodByte() const
+{
+    return static_cast<uint8_t>(CompressionMethodByte::RunLength);
+}
+
+UInt32 CompressionCodecRunLength::getMaxCompressedDataSize(UInt32 uncompressed_size) const
+{
+    return 1 + LZ4_COMPRESSBOUND(uncompressed_size);
```

Comment:
Can you add comments about why here is `1 + LZ4_...`?

Reply:
```
// If the data is not compressible as run-length encoding, we will compress it as LZ4.
// 1 byte for data type, and the rest for LZ4 compressed data.
```